pkg/utils/controller/placement_resolver.go

```diff
@@ -0,0 +1,85 @@
+/*
+Copyright 2025 The KubeFleet Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controller
+
+import (
+	"context"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	fleetv1beta1 "github.com/kubefleet-dev/kubefleet/apis/placement/v1beta1"
+	"github.com/kubefleet-dev/kubefleet/pkg/scheduler/queue"
+)
+
+const (
+	// namespaceSeparator is the separator used between namespace and name in placement keys
+	namespaceSeparator = "/"
+)
+
+// FetchPlacementFromKey resolves a PlacementKey to a concrete placement object that implements PlacementObj.
+func FetchPlacementFromKey(ctx context.Context, c client.Client, placementKey queue.PlacementKey) (fleetv1beta1.PlacementObj, error) {
+	keyStr := string(placementKey)
+
+	// Check if the key contains a namespace separator
+	if strings.Contains(keyStr, namespaceSeparator) {
+		// This is a namespaced ResourcePlacement
+		parts := strings.SplitN(keyStr, namespaceSeparator, 2)
```

Comment:
```suggestion
		parts := strings.SplitN(keyStr, namespaceSeparator, -1)
```

otherwise, when the string is "a/b/c", the result will be "a" and "b/c"